rllib/core/models/configs.py

```diff
@@ -46,38 +55,106 @@ def checked_build(self, framework, **kwargs):
     return decorator
 
 
+@ExperimentalAPI
+@dataclass
+class ModelConfig(abc.ABC):
+    """Base class for configuring a `Model` instance.
+
+    ModelConfigs are DL framework-agnostic.
+    A `Model` (as a sub-component of an `RLModule`) is built via calling the
+    respective ModelConfig's `build()` method.
+    RLModules build their sub-components this way after receiving one or more
+    `ModelConfig` instances from a Catalog object.
+
+    However, `ModelConfig` is not restricted to be used only with Catalog or RLModules.
+    Usage examples can be found in the individual Model classes', e.g.
+    see `ray.rllib.core.models.configs::MLPHeadConfig`.
+
+    Attributes:
+        input_dims: The input dimensions of the network
+        always_check_shapes: Whether to always check the inputs and outputs of the
+            model for the specifications. Input specifications are checked on failed
+            forward passes of the model regardless of this flag. If this flag is set
+            to `True`, inputs and outputs are checked on every call. This leads to
+            a slow-down and should only be used for debugging.
+        _output_dims: The output dimensions of the network. The value of this field
```

Comment:
Further down, this is called `output_dims`.
Also, if every modelconfig has output dims and we say that it's an attribute here, I believe we should put it back there with `input_dims` and `always_check_shapes `.

Reply:
Sorry, this should have been removed. `self.output_dims` is now just a property w/o private variable equivalent.
Fixed (removed the comment in the `Attributes:` list for this class).